dev-python/slimit/slimit-0.8.1.ebuild

```diff
@@ -0,0 +1,32 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python2_7 pypy )
+inherit distutils-r1
+
+DESCRIPTION="A JavaScript minifier written in Python"
+HOMEPAGE="https://slimit.readthedocs.io/en/latest/"
+SRC_URI="https://github.com/rspivak/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+
+RDEPEND="dev-python/ply[${PYTHON_USEDEP}]"
+DEPEND="${RDEPEND}
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	test? (
+		dev-python/pytest[${PYTHON_USEDEP}]
+		dev-python/pytest-runner[${PYTHON_USEDEP}]
+	)"
+
+python_prepare_all() {
+	distutils-r1_python_prepare_all
+}
+
+python_test() {
+	esetup.py pytest
```

Comment:
Why are you using pytest? FWICS it uses plain Python unittest, so please don't introduce unnecessary deps.

Reply:
That's what I thought, but if I run:
    
    paolop /var/tmp/portage/dev-python/slimit-0.8.1/work/slimit-0.8.1 # python2.7 ./setup.py test
    running test
    running egg_info
    creating src/slimit.egg-info
    writing requirements to src/slimit.egg-info/requires.txt
    writing src/slimit.egg-info/PKG-INFO
    writing top-level names to src/slimit.egg-info/top_level.txt
    writing dependency_links to src/slimit.egg-info/dependency_links.txt
    writing entry points to src/slimit.egg-info/entry_points.txt
    writing manifest file 'src/slimit.egg-info/SOURCES.txt'
    reading manifest file 'src/slimit.egg-info/SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    writing manifest file 'src/slimit.egg-info/SOURCES.txt'
    running build_ext
    
    ----------------------------------------------------------------------
    Ran 0 tests in 0.000s
    
    OK

no tests are run, while with:

    paolop /var/tmp/portage/dev-python/slimit-0.8.1/work/slimit-0.8.1 # python2.7 ./setup.py 
    pytest
    running pytest
    running egg_info
    writing requirements to src/slimit.egg-info/requires.txt
    writing src/slimit.egg-info/PKG-INFO
    writing top-level names to src/slimit.egg-info/top_level.txt
    writing dependency_links to src/slimit.egg-info/dependency_links.txt
    writing entry points to src/slimit.egg-info/entry_points.txt
    reading manifest file 'src/slimit.egg-info/SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    writing manifest file 'src/slimit.egg-info/SOURCES.txt'
    running build_ext
    ======================================================= test session starts ========================================================
    platform linux2 -- Python 2.7.14, pytest-3.2.2, py-1.4.34, pluggy-0.4.0
    rootdir: /var/tmp/portage/dev-python/slimit-0.8.1/work/slimit-0.8.1, inifile:
    plugins: virtualenv-1.2.11, shutil-1.2.11, cov-2.3.1, hypothesis-3.6.0, flaky-3.1.1, backports.unittest-mock-1.3
    collected 178 items                                                                                                                 
    
    src/slimit/tests/test_cmd.py .....
    src/slimit/tests/test_ecmavisitor.py ...................................................................
    src/slimit/tests/test_lexer.py ........................................
    src/slimit/tests/test_mangler.py ....
    src/slimit/tests/test_minifier.py ............................................
    src/slimit/tests/test_nodevisitor.py .
    src/slimit/tests/test_parser.py .................
    ===Flaky Test Report===
    
    
    ===End Flaky Test Report===
    
    =================================================== 178 passed in 441.70 seconds ===================================================
    
tests are run, and run correctly.

Reply:
Don't use setup.py.

https://wiki.gentoo.org/wiki/Project:Python/Tests

Reply:
I've read the wiki page several times over, and I've tried all the ways to run tests listed at https://wiki.gentoo.org/wiki/Project:Python/Tests#Test_suites. The only way that gives me any meaningful result is the one for pytest: nosetests yields several errors (and it's a far worst dependency than pytest, AFAIK), twisted-trial I couldn't find any way to make it work, unittest and unittest2 run no test at all.

I've thus modified the ebuild to run pytest directly (and added python3 support via a debian patch). It's the only way that does not imply restricting tests AFAICS. Running it this way loses multiprocessing support and is way slower, BTW, are you sure this is intended? Tell me how do you want me to proceed...

Tests must be run in ${BUILD_DIR} or python3 tests fail with:

    ImportError while importing test module '/var/tmp/portage/dev-python/slimit-0.8.1/work/slimit-0.8.1/src/slimit/tests/test_cmd.py'.
    Hint: make sure your test modules/packages have valid Python names.
    Traceback:
    src/slimit/__init__.py:27: in <module>
        from minifier import minify
    E   ModuleNotFoundError: No module named 'minifier'

Reply:
If it's that bad, then I guess using setup.py is justified.

Reply:
Ok. I've updated the ebuild to the best of my ability. Is it good enough to commit? Thanks to everybody for their support and guidance.